oak_functions/loader/src/logger.rs

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chrono::{SecondsFormat, Utc};
+use log::{Level, LevelFilter};
+use std::default::Default;
+
+/// A simple logger that splits logging between writing logs that contain only public, non-sensitive
+/// content and writing logs that could potentially contain sensitive content.
+///
+/// Writing of potentially sensitive content will be ignored unless the "oak-insecure" feature is
+/// enabled.
+#[derive(Clone)]
```

Comment:
Why is it cloned? I would expect that only references to the logger get passed around, or is that too complicated given the lifetime requirements of the current code?

Reply:
I initially used and `Arc<Logger>` everywhere, but I thought cloning will be more efficient as the struct contains only the log level filter. I am unsure how a single logger's lifetime will interact with the various async functions, but I will try it out and see.

Reply:
The lifetime is problematic in `create_and_start_server` with the `WasmHandler` clones in the closures. The compiler does not believe that `graceful_server.await` will outlive all `WasmHandler` instances. I tried using static lifetime annotations in various places, but that just moved the problem elsewhere.

I could use references everywhere except in the WasmHandler where it would use clones, but I am not sure whether it would be any better than just always cloning. Alternatively I could use something like `lazy_static!` to create a static singleton instance, but that feels a lot like moving back towards ambient authority.